### PR TITLE
Add command to add an auto-detected project as a favorite

### PR DESCRIPTION
**Pull request by ckaczor**
I have a bunch of auto-detected projects and I wanted a quick way to add some as a favorite.  This adds a context menu item that will quickly save a project as a favorite.

### CODE REVIEW

**Comment by alefragnani** [comment]
Hi @ckaczor ,

First of all, thanks for you PR and sorry for taking so long to answer. I was a bit busy in the last couple of weeks, with my vacation and the preparations for speaking at [Embarcadero Conference](www.embarcaderoconference.com.br).

About your PR, I think that instead of a new command, you should reuse the `Save Project` command, because it would still ask the user for the project name. 

Of course the `saveProject()` function itself should be updated to accept another path (today it uses `vscode.workspace.workspaceFolders[0]` only), but I think the changes would be fairly easy to do.

Could you update the PR for that?

Thank you

**Comment by ckaczor** [comment]
Thanks for the feedback - I updated the commit with the requested changes.

File: src/extension.ts [hunk 1]
```diff
@@ -251,21 +252,29 @@ export function activate(context: vscode.ExtensionContext) {
         }
     }
 
-    function saveProject() {
-        // Display a message box to the user
-        // let wpath = vscode.workspace.rootPath;
-        const workspace0 = vscode.workspace.workspaceFolders ? vscode.workspace.workspaceFolders[0] : undefined;
-        let wpath = workspace0 ? workspace0.uri.fsPath : undefined;
+    function saveProject(node: any = undefined) {
```
Comment (alefragnani) [inline src/extension.ts]: TSLint is complaining about this syntax: `no-unnecessary-initializer`. 
Use `node?: any` instead

**Comment by alefragnani** [comment]
Hi @ckaczor ,

I'm _finally_ returning to the PRs, and made two comments on that. Could you take a look? 

My intention is to start releasing updates in the weekend. 

Thank you

**Comment by ckaczor** [comment]
Changes should be all set - thanks for the feedback =)

**Pull request merged**